server/service/src/programs/program_indicator/query.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashMap;
+
+use repository::{
+    IndicatorColumnRow, IndicatorColumnRowRepository, IndicatorLineRow, IndicatorLineRowRepository,
+    Pagination, ProgramIndicatorFilter, ProgramIndicatorRepository, ProgramIndicatorRow,
+    ProgramIndicatorSort, RepositoryError, StorageConnection,
+};
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct IndicatorLine {
+    pub line: IndicatorLineRow,
+    pub columns: Vec<IndicatorColumnRow>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct ProgramIndicator {
+    pub program_indicator: ProgramIndicatorRow,
+    pub lines: Vec<IndicatorLine>,
+}
+
+pub fn program_indicators(
+    connection: &StorageConnection,
+    pagination: Pagination,
+    sort: Option<ProgramIndicatorSort>,
+    filter: Option<ProgramIndicatorFilter>,
+) -> Result<HashMap<String, ProgramIndicator>, RepositoryError> {
```

Comment:
HashMap probably not needed here though : )

Reply:
Good point!